src/WebJobs.Script.WebHost/WebHooks/WebJobsSdkExtensionHookProvider.cs

```diff
@@ -52,8 +61,30 @@ internal static Uri GetExtensionWebHookRoute(string name)
 
             bool isLocalhost = hostName.StartsWith("localhost:", StringComparison.OrdinalIgnoreCase);
```

Comment:
This code is not ran when tested locally:
`var hostName = settings.GetSetting(EnvironmentSettingNames.AzureWebsiteHostName);`
will return `null` 
if you **don't set your** `WEBSITE_HOSTNAME = localhost:12345` explicitly on your local env

We should consider `hostName == null` -->`localhost`